packages/cli/src/commands/source/catalog/add.ts

```diff
@@ -0,0 +1,160 @@
+import {UploadBatchCallbackData, CatalogSource} from '@coveo/push-api-client';
+import {Command, Flags, CliUx} from '@oclif/core';
+import {green} from 'chalk';
+import {
+  HasNecessaryCoveoPrivileges,
+  IsAuthenticated,
+  Preconditions,
+} from '../../../lib/decorators/preconditions';
+import {
+  readOrganizationPrivilege,
+  writeFieldsPrivilege,
+  writeSourceContentPrivilege,
+} from '../../../lib/decorators/preconditions/platformPrivilege';
+import {Trackable} from '../../../lib/decorators/preconditions/trackable';
+import {
+  withFile,
+  withCreateMissingFields,
+  withFolder,
+  withMaxConcurrent,
+} from '../../../lib/flags/sourceCommonFlags';
+import {AuthenticatedClient} from '../../../lib/platform/authenticatedClient';
+import {errorMessage, successMessage} from '../../../lib/push/userFeedback';
+import {getFileNames} from '../../../lib/utils/file';
+import {bold} from 'chalk';
+import dedent from 'ts-dedent';
+
+export default class SourceCatalogAdd extends Command {
+  public static description =
+    'Index a JSON document into a Coveo Catalog source. See https://docs.coveo.com/en/2956/coveo-for-commerce/index-commerce-catalog-content-with-the-stream-api for more information.';
+
+  public static flags = {
+    ...withFile(),
+    ...withFolder(),
+    ...withMaxConcurrent(),
+    ...withCreateMissingFields(),
+    fullUpload: Flags.boolean({
```

Comment:
By default, adding documents to a catalog source will trigger a [full document update](https://docs.coveo.com/en/l62e0540/coveo-for-commerce/full-document-update). This will act the same way as pushing stuff to a push source.

However, we recommend to perform an initial catalog update first. However, since this operation overwrites the entire source content, I was not confortable making this the default operation.

That's why I later check if the source contains items (line 87) to better guide the user.